big_tests/tests/rdbms_SUITE.erl

```diff
@@ -387,6 +394,48 @@ test_request_insert(Config) ->
                            selected_to_sorted(SelectResult))
       end, ok, #{name => request_queries}).
 
+test_wrapped_request(Config) ->
+    % given
+    erase_table(Config),
+    sql_prepare(Config, insert_one, test_types, [unicode],
+                <<"INSERT INTO test_types(unicode) VALUES (?)">>),
+    rpc(mim(), mongoose_metrics, ensure_metric, [global, [test_metric], histogram]),
+    WrapperFun = fun(SqlExecute) ->
+        {Time, Result} = timer:tc(SqlExecute),
+        mongoose_metrics:update(global, [test_metric], Time),
+        Result
+    end,
+
+    % when
+    sql_execute_wrapped_request(Config, insert_one, [<<"check1">>], WrapperFun),
+
+    % then
+    mongoose_helper:wait_until(
+        fun() ->
+            SelectResult = sql_query(Config, "SELECT unicode FROM test_types"),
+            ?assertEqual({selected, [{<<"check1">>}]}, selected_to_sorted(SelectResult))
+        end, ok, #{name => request_queries}),
+
+    {ok, Metric} = rpc(mim(), mongoose_metrics, get_metric_value, [global, [test_metric]]),
```

Comment:
there is a small chance (like really tiny, because SQL queries are pretty slow operation comparing to metric update operation) of a race condition (we first send a request and after update the metric).
Ideally you need to check a result of the request using `gen_server:wait_response` call before checking the metric value.